lib/kafka/consumer_interceptors.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Kafka
+  # Holds a list of Kafka::ConsumerInterceptor
+  # and wraps calls to a chain of custom consumer interceptors.
+  class ConsumerInterceptors
+    def initialize(interceptors:, logger:)
+      @interceptors = interceptors
+      @logger = TaggedLogger.new(logger)
+    end
+
+    # This is called from Kafka::Consumer once the batches are fetched.
+    #
+    # @param batches [Array<Kafka::FetchedMessage>]
+    # @return [Array<Kafka::FetchedMessage>] intercepted batches.
+    def on_consume(batches:)
+      intercepted_batches = batches
+      @interceptors.each do |interceptor|
+        begin
+          intercepted_batches = interceptor.on_consume(batches: intercepted_batches)
+        rescue Exception => e
+          @logger.warn "Error executing interceptor on_consume #{e.class}: #{e.message}\n#{e.backtrace.join("\n")}"
+        end
+      end
+
+      intercepted_batches
+    end
+
+    # Closes every interceptor.
+    # @return [nil]
+    def close
+      @interceptors.each do |interceptor|
```

Comment:
it seems we'll need to check for`nil` here, `initialize` doesn't guarantee a value here

Reply:
we cannot guarantee a value because all exceptions in the interceptors must be caught